src/AWS.Distro.OpenTelemetry.AutoInstrumentation/AwsMetricAttributeGenerator.cs

```diff
@@ -408,8 +413,32 @@ private static void SetRemoteResourceTypeAndIdentifier(Activity span, ActivityTa
                 remoteResourceType = NormalizedSQSServiceName + "::Queue";
                 remoteResourceIdentifier = EscapeDelimiters(GetQueueName((string?)span.GetTagItem(AttributeAWSSQSQueueUrl)));
             }
-        }
-        else if (IsDBSpan(span))
+            else if (IsKeyPresent(span, AttributeAWSBedrockGuardrailId))
+            {
+                remoteResourceType = NormalizedBedrockServiceName + "::Guardrail";
+                remoteResourceIdentifier = EscapeDelimiters((string?)span.GetTagItem(AttributeAWSBedrockGuardrailId));
+            }
+            else if (IsKeyPresent(span, AttributeGenAiModelId))
+            {
+                remoteResourceType = NormalizedBedrockServiceName + "::Model";
+                remoteResourceIdentifier = EscapeDelimiters((string?)span.GetTagItem(AttributeGenAiModelId));
+            }
+            else if (IsKeyPresent(span, AttributeAWSBedrockAgentId))
+            {
+                remoteResourceType = NormalizedBedrockServiceName + "::Agent";
+                remoteResourceIdentifier = EscapeDelimiters((string?)span.GetTagItem(AttributeAWSBedrockAgentId));
+            }
+            else if (IsKeyPresent(span, AttributeAWSBedrockKnowledgeBaseId))
+            {
+                remoteResourceType = NormalizedBedrockServiceName + "::KnowledgeBase";
+                remoteResourceIdentifier = EscapeDelimiters((string?)span.GetTagItem(AttributeAWSBedrockKnowledgeBaseId));
+            }
+            else if (IsKeyPresent(span, AttributeAWSBedrockDataSourceId))
+            {
+                remoteResourceType = NormalizedBedrockServiceName + "::DataSource";
+                remoteResourceIdentifier = EscapeDelimiters((string?)span.GetTagItem(AttributeAWSBedrockDataSourceId));
+            }
+        } else if (IsDBSpan(span))
```

Comment:
Please address `Check warning on line 441 in src/AWS.Distro.OpenTelemetry.AutoInstrumentation/AwsMetricAttributeGenerator.cs`

Reply:
fixed